### ISSUE TITLE
installable as Python package: the entry-point clashes with the script?

### ISSUE THREAD

**Issue by imz** [nicktimko/autolycus#1]
I saw your "installable" branch for hg-fast-export at github. I thought of using it, since installing a Python package seems a better way to go (so that it is re-usable from other Python scripts).

But I'm doubt about your decision about [the name for the entry-points](https://github.com/nicktimko/fast-export/blob/7c704dbf9db6c4287b397edfa9b05a36af4c8051/setup.py#L18): there is also the `hg-fast-export.sh` script, which people usually invoke having it installed simply as `/usr/bin/hg-fast-export`. Won't your names for the entry-points, e.g., `hg-fast-export`, clash with the usual filename of this script?

(As I've learned from a reply by email, this is not final and the author of `setup.py` is open to suggestions.)

**Comment by nicktimko** [comment]
Yeah, as mentioned in that email (I didn't have issues enabled here for some reason), my names are kinda all over the place (`asdf`, Hermes, etc.) and open to suggestion. More about getting the functionality repackaged right now, and mostly for the Hg to Git exporter/updater.

Need to move some logic in the `.sh` to the `.py` it seems (bunch of required arguments in the `.py` that are auto-defaulted by the `.sh`)

**Comment by nicktimko** [comment]
Shuffled around the names a bit. Random Wikipedia hole gave me "Autolycus", son of Hermes (Mercury) who's the "Prince of Thieves". Mercury...ial...stealing/gitting stuff... whatever.

Anyways, diving more and more into it, despite it having a Python component, the script for that is mostly a wrapper to format the objects coming out of Mercurial, the main plumbing is done by the shell script. I'm trying to reverse engineer a bit to get it functional again. It needs to be Python so it's Windows compatible, which is doable, I just [don't know what this](https://github.com/nicktimko/autolycus/blob/master/shell-scripts/hg-fast-export.sh#L101):

``` bash
exec 3>&1
{ read -r _err1 || :; read -r _err2 || :; } <<-EOT
$(
  exec 4>&3 3>&1 1>&4 4>&-
  {
    _e1=0
...
```

is even doing. Looks like pipe hell.
